x/ccv/provider/keeper/proposal_test.go

```diff
@@ -254,65 +256,118 @@ func TestPendingConsumerAdditionPropDeletion(t *testing.T) {
 	}
 }
 
-// TestPendingConsumerAdditionPropOrder tests that pending consumer addition proposals
-// are accessed in order by timestamp via the iterator
-func TestPendingConsumerAdditionPropOrder(t *testing.T) {
+// TestGetConsumerAdditionPropsToExecute tests that pending consumer addition proposals
+// that are ready to execute are accessed in order by timestamp via the iterator
+func TestGetConsumerAdditionPropsToExecute(t *testing.T) {
 
 	now := time.Now().UTC()
-
-	// props with unique chain ids and spawn times
-	sampleProp1 := providertypes.ConsumerAdditionProposal{ChainId: "1", SpawnTime: now}
-	sampleProp2 := providertypes.ConsumerAdditionProposal{ChainId: "2", SpawnTime: now.Add(1 * time.Hour)}
-	sampleProp3 := providertypes.ConsumerAdditionProposal{ChainId: "3", SpawnTime: now.Add(2 * time.Hour)}
-	sampleProp4 := providertypes.ConsumerAdditionProposal{ChainId: "4", SpawnTime: now.Add(3 * time.Hour)}
-	sampleProp5 := providertypes.ConsumerAdditionProposal{ChainId: "5", SpawnTime: now.Add(4 * time.Hour)}
+	sampleProp1 := providertypes.ConsumerAdditionProposal{ChainId: "chain-2", SpawnTime: now}
+	sampleProp2 := providertypes.ConsumerAdditionProposal{ChainId: "chain-1", SpawnTime: now.Add(time.Hour)}
+	sampleProp3 := providertypes.ConsumerAdditionProposal{ChainId: "chain-4", SpawnTime: now.Add(-time.Hour)}
+	sampleProp4 := providertypes.ConsumerAdditionProposal{ChainId: "chain-3", SpawnTime: now}
+	sampleProp5 := providertypes.ConsumerAdditionProposal{ChainId: "chain-1", SpawnTime: now.Add(2 * time.Hour)}
+
+	getExpectedOrder := func(props []providertypes.ConsumerAdditionProposal, accessTime time.Time) []providertypes.ConsumerAdditionProposal {
```

Comment:
Not a huge fan of unit tests defined like this, the old way of hardcoding `expectedOrderedProps` for each test case is more effective.

In order for a quick reader to gain context, they have to not only understand the business logic of the software being tested (`GetConsumerAdditionPropsToExecute` in this case), but also the logic defined in `getExpectedOrder`.

The logic you've written for `getExpectedOrder` looks correct. But if we start writing unit tests with this pattern, it encourages devs to redefine the logic they're testing in the first place. Functional black-box testing (hardcoded expectations) is most effective for unit tests imo. 

Functional testing becomes less relevant for higher level tests, I'm only talking about UTs here.